File: ScaleformUI_Csharp/PauseMenu/Items/KeymapItem.cs
using CitizenFX.Core;
using CitizenFX.Core.Native;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ScaleformUI.PauseMenu
{
    public class KeymapItem : BasicTabItem
    {
        public string PrimaryKeyboard;
        public string PrimaryGamepad;
        public string SecondaryKeyboard;
        public string SecondaryGamepad;
        public KeymapItem(string title, string primaryKeyboard) : this(title, primaryKeyboard, "") { }
        public KeymapItem(string title, string primaryKeyboard, string secondaryKeyboard) : base(title)
        {
            PrimaryKeyboard = primaryKeyboard;
            PrimaryGamepad = primaryKeyboard;
            SecondaryKeyboard = secondaryKeyboard;
            SecondaryGamepad = secondaryKeyboard;
        }
        public KeymapItem(string title, string primaryKeyboard, string primaryGamepad, string secondaryKeyboard, string secondaryGamepad) : base(title)
        {
            PrimaryKeyboard = primaryKeyboard;
            PrimaryGamepad = primaryGamepad;
            SecondaryKeyboard = secondaryKeyboard;
            SecondaryGamepad = secondaryGamepad;
        }
    }
}

File: ScaleformUI_Lua/src/PauseMenu/Items/KeymapItem.lua
KeymapItem = setmetatable({}, KeymapItem)
KeymapItem.__index = KeymapItem
KeymapItem.__call = function()
    return "BasicTabItem", "KeymapItem"
end

---@class KeymapItem
---@field public Label string
---@field public PrimaryKeyboard string
---@field public PrimaryGamepad string
---@field public SecondaryKeyboard string
---@field public SecondaryGamepad string

---Creates a new KeymapItem.
---@param title string
---@param primaryKeyboard string|nil
---@param primaryGamepad string|nil
---@param secondaryKeyboard string|nil
---@param secondaryGamepad string|nil
---@return table
function KeymapItem.New(title, primaryKeyboard, primaryGamepad, secondaryKeyboard, secondaryGamepad)
    local data = {}
    if secondaryKeyboard == nil and secondaryGamepad == nil then
        data = {
            Label = title,
            PrimaryKeyboard = primaryKeyboard,
            PrimaryGamepad = primaryGamepad,
            SecondaryKeyboard = "",
            SecondaryGamepad = "",
        }
    else
        data = {
            Label = title,
            PrimaryKeyboard = primaryKeyboard or "",
            PrimaryGamepad = primaryGamepad or "",
            SecondaryKeyboard = secondaryKeyboard or "",
            SecondaryGamepad = secondaryGamepad or "",
        }
    end
    return setmetatable(data, KeymapItem)
end
